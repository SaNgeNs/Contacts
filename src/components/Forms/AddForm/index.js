import withErrorHandler from 'Components/withErrorHandler';
import loadable from '@loadable/component';
import { reduxForm } from 'redux-form';

const AddForm = loadable(() => import(/* webpackChunkName: "add_form_component" */'./AddForm'));

export default withErrorHandler(reduxForm({
  form: 'createCard',
  initialValues: {
    name: '',
    phone: '',
    email: '',
  },
})(AddForm));
