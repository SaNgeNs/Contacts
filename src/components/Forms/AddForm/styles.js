export const styles = {
  title: {
    fontSize: '20px',
    margin: '10px',
  },
};

export default styles;
