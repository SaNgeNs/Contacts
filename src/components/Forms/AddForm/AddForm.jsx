import React, { memo } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import { useDispatch } from 'react-redux';
import addCard from 'Actions/addCard';

import FieldInput from 'Components/FieldInput';
import validation from 'Utils/validation';
import styles from './styles';

export const AddForm = ({
  classes,
  handleSubmit,
  reset
}) => {
  const dispatch = useDispatch();

  const submitForm = (values) => {
    validation(values);

    dispatch(addCard({
      ...values,
      id: Date.now(),
      isFavourite: false,
    }));
    dispatch(reset('createCard'));
  };
  
  return (
    <form
      noValidate
      onSubmit={handleSubmit(submitForm)}
    >
      <h1 className={classes.title}>Add Contact:</h1>

      <FieldInput name="name" id="name" label="name" type="text" variant="outlined" />
      <FieldInput normalize={(val) => val.replace(/\D+/g,"").slice(0, 9)} name="phone" id="phone" label="phone" type="tel" variant="outlined" />
      <FieldInput name="email" id="email" label="email" type="email" variant="outlined" />

      <Button
        type="submit"
        variant="contained"
        color="primary"
        size="large"
        startIcon={<AddIcon />}
      >
        Add
      </Button>
    </form>
  );
};

export default memo(withStyles(styles)(AddForm));
