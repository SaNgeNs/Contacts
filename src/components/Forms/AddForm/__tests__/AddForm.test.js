import React from 'react';
import { fireEvent } from '@testing-library/react';
import customRender from 'Mock/customRender';
import { reduxForm } from 'redux-form';

import AddForm from '../AddForm';

describe('<AddForm />', () => {
  it('should render add form', () => {
    const Component = reduxForm({
      form: 'createCard',
      initialValues: {
        name: '',
        phone: '',
        email: '',
      },
    })(AddForm);

    const { container, unmount } = customRender(
      <Component />,
    );

    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should clear input field after submit', () => {
    const Component = reduxForm({
      form: 'createCard',
      initialValues: {
        name: 'qweqwt',
        phone: '123456789',
        email: 'asd@masdil.ru',
      },
    })(AddForm);

    const { container, unmount } = customRender(
      <Component />,
    );

    const button = container.querySelector('[type="submit"]');

    fireEvent.click(button);

    expect(container).toMatchSnapshot();
    unmount();
  });
});
