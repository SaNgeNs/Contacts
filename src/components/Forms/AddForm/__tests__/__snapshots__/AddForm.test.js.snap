// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddForm /> should clear input field after submit 1`] = `
<div>
  <form
    novalidate=""
  >
    <h1
      class="AddForm-title-2"
    >
      Add Contact:
    </h1>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeLarge"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Add
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;

exports[`<AddForm /> should render add form 1`] = `
<div>
  <form
    novalidate=""
  >
    <h1
      class="AddForm-title-1"
    >
      Add Contact:
    </h1>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeLarge"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Add
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
